ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/cio/NettyResponsePipeline.kt

```diff
@@ -152,161 +108,173 @@ internal class NettyResponsePipeline constructor(
         call.responseWriteJob.cancel()
         call.response.cancel()
         call.dispose()
-        responses.cancel()
-        requestQueue.cancel()
+        call.callFinished.setFailure(t)
     }
 
-    private fun processUpgrade(responseMessage: Any): ChannelFuture {
-        val future = dst.write(responseMessage)
-        encapsulation.upgrade(dst)
-        encapsulation = WriterEncapsulation.Raw
-        dst.flush()
+    private fun processUpgrade(call: NettyApplicationCall, responseMessage: Any): ChannelFuture {
```

Comment:
Nit: I would prefer `processUpgradeResponse` or `processHttpUpgrade`

Reply:
Renamed to `processUpgradeResponse`